src/main/java/com/cloudogu/smeagol/wiki/infrastructure/PageController.java

```diff
@@ -0,0 +1,114 @@
+package com.cloudogu.smeagol.wiki.infrastructure;
+
+import com.cloudogu.smeagol.wiki.domain.*;
+import com.cloudogu.smeagol.wiki.usecase.*;
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import de.triology.cb.CommandBus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+
+@RestController
+@RequestMapping(PageController.MAPPING)
+public class PageController {
+
+    static final String MAPPING = "/api/v1/repositories/{repositoryId}/branches/{branch}/pages";
+
+    private final PageResourceAssembler assembler = new PageResourceAssembler();
+
+    private final PageRepository repository;
+    private final WildcardPathExtractor pathExtractor;
+    private final CommandBus commandBus;
+
+    @Autowired
+    public PageController(PageRepository repository, WildcardPathExtractor pathExtractor, CommandBus commandBus) {
+        this.repository = repository;
+        this.pathExtractor = pathExtractor;
+        this.commandBus = commandBus;
+    }
+
+    @RequestMapping("**")
+    public ResponseEntity<PageResource> findByWikiIdAndPath(
+            HttpServletRequest request,
+            @PathVariable("repositoryId") String repositoryId,
+            @PathVariable("branch") String branch
+    ) {
+        WikiId id = new WikiId(repositoryId, branch);
+        Path path = pathExtractor.extractPathFromRequest(request, MAPPING, id);
+
+        Optional<Page> byWikiIdAndPath = repository.findByWikiIdAndPath(id, path);
+        if (byWikiIdAndPath.isPresent()) {
+            return ResponseEntity.ok(assembler.toResource(byWikiIdAndPath.get()));
+        }
+        return ResponseEntity.notFound().build();
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value = "**")
+    public ResponseEntity<Void> createOrEdit(
+            HttpServletRequest request,
+            @PathVariable("repositoryId") String repositoryId,
+            @PathVariable("branch") String branch,
+            @RequestBody CreateOrEditRequestPayload payload
+    ) throws URISyntaxException {
+        WikiId id = new WikiId(repositoryId, branch);
+        Path path = pathExtractor.extractPathFromRequest(request, MAPPING, id);
+
+        // TODO return new page? this would safe us one request from the frontend. Is this resty?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1135)